pennylane/optimize/qng.py

```diff
@@ -204,8 +204,15 @@ def step_and_cost(
 
                 metric_tensor_fn = qml.metric_tensor(qnode, approx=self.approx)
 
-            self.metric_tensor = metric_tensor_fn(*args, **kwargs)
-            self.metric_tensor += self.lam * np.identity(self.metric_tensor.shape[0])
+            _metric_tensor = metric_tensor_fn(*args, **kwargs)
+            # Reshape metric tensor to be square
+            shape = qml.math.shape(_metric_tensor)
+            size = qml.math.prod(shape[: len(shape) // 2])
+            self.metric_tensor = qml.math.reshape(_metric_tensor, (size, size))
```

Comment:
does reshaping the metric tensor to be square cause issues later on when applying it to the gradient vector (or is the gradient vector always flat, even if the weights aren't?)

An alternative would be to reshape the identity matrix to match the shape of the metric tensor

Reply:
Yeah, `apply_grad` is a "flat" method. I started looking into this but then noted that the entire `optimize` submodule is neither interface-agnostic nor ready to tackle anything but flat arrays :sweat_smile: So I decided to not go further here, but to at least incorporate this change already. One could argue that this is inconsistent and instead postpone this change to some `optimize` updating PR. I think my motivation here was that some tests were failing as it was.

Reply:
Sounds good :+1:

We have a goal to expand the optimizers: #1676 #846, but it is a very non-trivial thing to design